app/process/recovery/recovery.go

```diff
@@ -159,23 +157,30 @@ func (r Recovery) transfersRecovery(from int64, to int64) error {
 	for _, tx := range txns {
 		amount, asset, err := tx.GetIncomingTransfer(r.accountID.String())
 		if err != nil {
-			r.logger.Errorf("Skipping recovery of TX [%s]. Invalid amount. Error: [%s]", tx.TransactionID, err)
+			r.logger.Errorf("[%s] - Skipping recovery. Invalid amount. Error: [%s]", tx.TransactionID, err)
 			continue
 		}
+
+		valid, targetAsset := r.contracts.IsValidBridgeAsset(asset)
```

Comment:
Idea: Change the interface for retrieving the target asset to `r.contracts.ParseSourceAsset` or something similar and for it to return directly the asset. If an error is returned, we can think of it as the asset is unsupported